[ ERROR ] - [ edu.hitsz.commons.utils.Parser.parserServerConfig(Parser.java:69) ] - 2024-10-10 13:13:42  
读取serverconfig.xml失败
java.io.FileNotFoundException: \home\gomoku\serverconfig.xml (The system cannot find the path specified)
	at java.base/java.io.FileInputStream.open0(Native Method)
	at java.base/java.io.FileInputStream.open(FileInputStream.java:213)
	at java.base/java.io.FileInputStream.<init>(FileInputStream.java:152)
	at java.base/java.io.FileInputStream.<init>(FileInputStream.java:106)
	at java.base/java.io.FileReader.<init>(FileReader.java:60)
	at edu.hitsz.commons.utils.XMLParseUtil.createReadRoot(XMLParseUtil.java:48)
	at edu.hitsz.commons.utils.Parser.parserServerConfig(Parser.java:60)
	at edu.hitsz.cluster.server.Server.initConfig(Server.java:131)
	at edu.hitsz.cluster.server.Server.init(Server.java:100)
	at edu.hitsz.cluster.server.Server.start(Server.java:43)
	at edu.hitsz.driver.ServerDriver.main(ServerDriver.java:11)
[ ERROR ] - [ edu.hitsz.cluster.server.Server.init(Server.java:107) ] - 2024-10-10 13:13:42  
Server初始化失败!
[ ERROR ] - [ edu.hitsz.cluster.server.Server.startRemoting(Server.java:88) ] - 2024-10-10 13:13:42  
Server通信服务开启失败!
java.lang.NullPointerException: Cannot invoke "edu.hitsz.remoting.delegate.RemotingServerDelegate.start()" because "this.remotingServer" is null
	at edu.hitsz.cluster.server.Server.startRemoting(Server.java:79)
	at edu.hitsz.cluster.server.Server.start(Server.java:44)
	at edu.hitsz.driver.ServerDriver.main(ServerDriver.java:11)
[ INFO ] - [ io.netty.util.internal.PlatformDependent.<clinit>(PlatformDependent.java:80) ] - 2024-10-10 13:41:47  
Your platform does not provide complete low-level API for accessing direct buffers reliably. Unless explicitly requested, heap buffer will always be preferred to avoid potential system unstability.
[ INFO ] - [ edu.hitsz.remoting.AbstractRemoting$RemotingEventExecutor.run(AbstractRemoting.java:502) ] - 2024-10-10 13:41:48  
RemotingEventExecutor service started
[ INFO ] - [ io.netty.util.internal.PlatformDependent.<clinit>(PlatformDependent.java:80) ] - 2024-10-10 13:50:32  
Your platform does not provide complete low-level API for accessing direct buffers reliably. Unless explicitly requested, heap buffer will always be preferred to avoid potential system unstability.
[ INFO ] - [ edu.hitsz.remoting.AbstractRemoting$RemotingEventExecutor.run(AbstractRemoting.java:502) ] - 2024-10-10 13:50:32  
RemotingEventExecutor service started
[ INFO ] - [ io.netty.util.internal.PlatformDependent.<clinit>(PlatformDependent.java:80) ] - 2024-10-10 13:52:41  
Your platform does not provide complete low-level API for accessing direct buffers reliably. Unless explicitly requested, heap buffer will always be preferred to avoid potential system unstability.
[ INFO ] - [ io.netty.util.internal.PlatformDependent.<clinit>(PlatformDependent.java:80) ] - 2024-10-10 13:58:41  
Your platform does not provide complete low-level API for accessing direct buffers reliably. Unless explicitly requested, heap buffer will always be preferred to avoid potential system unstability.
[ INFO ] - [ edu.hitsz.remoting.AbstractRemoting$RemotingEventExecutor.run(AbstractRemoting.java:502) ] - 2024-10-10 13:58:41  
RemotingEventExecutor service started
[ INFO ] - [ io.netty.util.internal.PlatformDependent.<clinit>(PlatformDependent.java:80) ] - 2024-10-10 14:04:32  
Your platform does not provide complete low-level API for accessing direct buffers reliably. Unless explicitly requested, heap buffer will always be preferred to avoid potential system unstability.
[ INFO ] - [ edu.hitsz.remoting.AbstractRemoting$RemotingEventExecutor.run(AbstractRemoting.java:502) ] - 2024-10-10 14:04:33  
RemotingEventExecutor service started
[ INFO ] - [ io.netty.util.internal.PlatformDependent.<clinit>(PlatformDependent.java:80) ] - 2024-10-10 14:24:23  
Your platform does not provide complete low-level API for accessing direct buffers reliably. Unless explicitly requested, heap buffer will always be preferred to avoid potential system unstability.
[ INFO ] - [ edu.hitsz.remoting.AbstractRemoting$RemotingEventExecutor.run(AbstractRemoting.java:502) ] - 2024-10-10 14:24:24  
RemotingEventExecutor service started
[ INFO ] - [ edu.hitsz.remoting.AbstractRemotingClient.createChannel(AbstractRemotingClient.java:185) ] - 2024-10-10 14:43:44  
开始尝试建立通信连接:localhost:8888
[ INFO ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.channelRegistered(NettyRemotingServer.java:116) ] - 2024-10-10 14:43:44  
channel注册成功:127.0.0.1:53812
[ INFO ] - [ edu.hitsz.remoting.AbstractRemotingClient.createChannel(AbstractRemotingClient.java:204) ] - 2024-10-10 14:43:44  
建立远程连接成功:localhost:8888,edu.hitsz.remoting.netty.NettyChannelFuture@2721ae91
[ INFO ] - [ io.netty.util.internal.PlatformDependent.<clinit>(PlatformDependent.java:80) ] - 2024-10-10 14:51:56  
Your platform does not provide complete low-level API for accessing direct buffers reliably. Unless explicitly requested, heap buffer will always be preferred to avoid potential system unstability.
[ ERROR ] - [ edu.hitsz.remoting.netty.NettyRemotingServer.serverStart(NettyRemotingServer.java:78) ] - 2024-10-10 14:51:57  
Netty通信服务Server端启动失败!
java.net.BindException: Address already in use: bind
	at java.base/sun.nio.ch.Net.bind0(Native Method)
	at java.base/sun.nio.ch.Net.bind(Net.java:565)
	at java.base/sun.nio.ch.ServerSocketChannelImpl.netBind(ServerSocketChannelImpl.java:344)
	at java.base/sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:301)
	at java.base/sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:89)
	at io.netty.channel.socket.nio.NioServerSocketChannel.doBind(NioServerSocketChannel.java:102)
	at io.netty.channel.AbstractChannel$AbstractUnsafe.bind(AbstractChannel.java:479)
	at io.netty.channel.DefaultChannelPipeline$HeadHandler.bind(DefaultChannelPipeline.java:1000)
	at io.netty.channel.DefaultChannelHandlerContext.invokeBind(DefaultChannelHandlerContext.java:457)
	at io.netty.channel.DefaultChannelHandlerContext.bind(DefaultChannelHandlerContext.java:442)
	at io.netty.channel.DefaultChannelPipeline.bind(DefaultChannelPipeline.java:842)
	at io.netty.channel.AbstractChannel.bind(AbstractChannel.java:194)
	at io.netty.bootstrap.AbstractBootstrap$2.run(AbstractBootstrap.java:331)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:354)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:353)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:101)
	at java.base/java.lang.Thread.run(Thread.java:1583)
[ INFO ] - [ edu.hitsz.remoting.AbstractRemoting$RemotingEventExecutor.run(AbstractRemoting.java:502) ] - 2024-10-10 14:51:57  
RemotingEventExecutor service started
[ INFO ] - [ edu.hitsz.remoting.AbstractRemotingClient.createChannel(AbstractRemotingClient.java:185) ] - 2024-10-10 14:52:05  
开始尝试建立通信连接:localhost:8888
[ INFO ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.channelRegistered(NettyRemotingServer.java:116) ] - 2024-10-10 14:52:05  
channel注册成功:127.0.0.1:53922
[ INFO ] - [ edu.hitsz.remoting.AbstractRemotingClient.createChannel(AbstractRemotingClient.java:204) ] - 2024-10-10 14:52:05  
建立远程连接成功:localhost:8888,edu.hitsz.remoting.netty.NettyChannelFuture@84d7833
[ WARN ] - [ io.netty.channel.DefaultChannelPipeline$TailHandler.exceptionCaught(DefaultChannelPipeline.java:958) ] - 2024-10-10 14:59:40  
An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
java.net.SocketException: Connection reset
	at java.base/sun.nio.ch.SocketChannelImpl.throwConnectionReset(SocketChannelImpl.java:401)
	at java.base/sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:434)
	at io.netty.buffer.UnpooledHeapByteBuf.setBytes(UnpooledHeapByteBuf.java:256)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:871)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:208)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:115)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:485)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:417)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:348)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:101)
	at java.base/java.lang.Thread.run(Thread.java:1583)
[ INFO ] - [ io.netty.util.internal.PlatformDependent.<clinit>(PlatformDependent.java:80) ] - 2024-10-10 15:07:38  
Your platform does not provide complete low-level API for accessing direct buffers reliably. Unless explicitly requested, heap buffer will always be preferred to avoid potential system unstability.
[ INFO ] - [ edu.hitsz.remoting.AbstractRemoting$RemotingEventExecutor.run(AbstractRemoting.java:502) ] - 2024-10-10 15:07:39  
RemotingEventExecutor service started
[ INFO ] - [ edu.hitsz.remoting.AbstractRemotingClient.createChannel(AbstractRemotingClient.java:185) ] - 2024-10-10 15:07:46  
开始尝试建立通信连接:localhost:8888
[ INFO ] - [ edu.hitsz.remoting.AbstractRemotingClient.createChannel(AbstractRemotingClient.java:204) ] - 2024-10-10 15:07:46  
建立远程连接成功:localhost:8888,edu.hitsz.remoting.netty.NettyChannelFuture@6378e52
[ INFO ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.channelRegistered(NettyRemotingServer.java:116) ] - 2024-10-10 15:07:46  
channel注册成功:127.0.0.1:54043
[ INFO ] - [ io.netty.util.internal.PlatformDependent.<clinit>(PlatformDependent.java:80) ] - 2024-10-10 15:08:18  
Your platform does not provide complete low-level API for accessing direct buffers reliably. Unless explicitly requested, heap buffer will always be preferred to avoid potential system unstability.
[ ERROR ] - [ edu.hitsz.remoting.netty.NettyRemotingServer.serverStart(NettyRemotingServer.java:78) ] - 2024-10-10 15:08:18  
Netty通信服务Server端启动失败!
java.net.BindException: Address already in use: bind
	at java.base/sun.nio.ch.Net.bind0(Native Method)
	at java.base/sun.nio.ch.Net.bind(Net.java:565)
	at java.base/sun.nio.ch.ServerSocketChannelImpl.netBind(ServerSocketChannelImpl.java:344)
	at java.base/sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:301)
	at java.base/sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:89)
	at io.netty.channel.socket.nio.NioServerSocketChannel.doBind(NioServerSocketChannel.java:102)
	at io.netty.channel.AbstractChannel$AbstractUnsafe.bind(AbstractChannel.java:479)
	at io.netty.channel.DefaultChannelPipeline$HeadHandler.bind(DefaultChannelPipeline.java:1000)
	at io.netty.channel.DefaultChannelHandlerContext.invokeBind(DefaultChannelHandlerContext.java:457)
	at io.netty.channel.DefaultChannelHandlerContext.bind(DefaultChannelHandlerContext.java:442)
	at io.netty.channel.DefaultChannelPipeline.bind(DefaultChannelPipeline.java:842)
	at io.netty.channel.AbstractChannel.bind(AbstractChannel.java:194)
	at io.netty.bootstrap.AbstractBootstrap$2.run(AbstractBootstrap.java:331)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:354)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:353)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:101)
	at java.base/java.lang.Thread.run(Thread.java:1583)
[ INFO ] - [ edu.hitsz.remoting.AbstractRemoting$RemotingEventExecutor.run(AbstractRemoting.java:502) ] - 2024-10-10 15:08:18  
RemotingEventExecutor service started
[ INFO ] - [ edu.hitsz.remoting.AbstractRemotingClient.createChannel(AbstractRemotingClient.java:185) ] - 2024-10-10 15:08:25  
开始尝试建立通信连接:localhost:8888
[ INFO ] - [ edu.hitsz.remoting.AbstractRemotingClient.createChannel(AbstractRemotingClient.java:204) ] - 2024-10-10 15:08:25  
建立远程连接成功:localhost:8888,edu.hitsz.remoting.netty.NettyChannelFuture@379c9cf1
[ INFO ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.channelRegistered(NettyRemotingServer.java:116) ] - 2024-10-10 15:08:25  
channel注册成功:127.0.0.1:54052
[ WARN ] - [ io.netty.channel.DefaultChannelPipeline$TailHandler.exceptionCaught(DefaultChannelPipeline.java:958) ] - 2024-10-10 15:18:24  
An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
java.net.SocketException: Connection reset
	at java.base/sun.nio.ch.SocketChannelImpl.throwConnectionReset(SocketChannelImpl.java:401)
	at java.base/sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:434)
	at io.netty.buffer.UnpooledHeapByteBuf.setBytes(UnpooledHeapByteBuf.java:256)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:871)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:208)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:115)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:485)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:417)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:348)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:101)
	at java.base/java.lang.Thread.run(Thread.java:1583)
[ WARN ] - [ io.netty.bootstrap.AbstractBootstrap.setChannelOption(AbstractBootstrap.java:464) ] - 2024-10-10 15:30:23  
Unknown channel option 'SO_KEEPALIVE' for channel '[id: 0xef568dea]'
[ INFO ] - [ edu.hitsz.remoting.AbstractRemoting$RemotingEventExecutor.run(AbstractRemoting.java:502) ] - 2024-10-10 15:30:23  
RemotingEventExecutor service started
[ INFO ] - [ edu.hitsz.remoting.AbstractRemotingClient.createChannel(AbstractRemotingClient.java:185) ] - 2024-10-10 15:31:02  
开始尝试建立通信连接:localhost:8888
[ INFO ] - [ edu.hitsz.remoting.AbstractRemotingClient.createChannel(AbstractRemotingClient.java:204) ] - 2024-10-10 15:31:02  
建立远程连接成功:localhost:8888,edu.hitsz.remoting.netty.NettyChannelFuture@5670d0f2
[ INFO ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.channelRegistered(NettyRemotingServer.java:116) ] - 2024-10-10 15:31:02  
channel注册成功:127.0.0.1:54336
[ INFO ] - [ edu.hitsz.remoting.AbstractRemotingClient.createChannel(AbstractRemotingClient.java:185) ] - 2024-10-10 15:31:30  
开始尝试建立通信连接:localhost:8888
[ INFO ] - [ edu.hitsz.remoting.AbstractRemotingClient.createChannel(AbstractRemotingClient.java:204) ] - 2024-10-10 15:31:30  
建立远程连接成功:localhost:8888,edu.hitsz.remoting.netty.NettyChannelFuture@31f98395
[ INFO ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.channelRegistered(NettyRemotingServer.java:116) ] - 2024-10-10 15:31:30  
channel注册成功:127.0.0.1:54342
[ INFO ] - [ edu.hitsz.remoting.AbstractRemotingClient.createChannel(AbstractRemotingClient.java:185) ] - 2024-10-10 15:31:50  
开始尝试建立通信连接:localhost:8888
[ INFO ] - [ edu.hitsz.remoting.AbstractRemotingClient.createChannel(AbstractRemotingClient.java:204) ] - 2024-10-10 15:31:50  
建立远程连接成功:localhost:8888,edu.hitsz.remoting.netty.NettyChannelFuture@33fb627d
[ INFO ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.channelRegistered(NettyRemotingServer.java:116) ] - 2024-10-10 15:31:50  
channel注册成功:127.0.0.1:54344
[ WARN ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.exceptionCaught(NettyRemotingServer.java:150) ] - 2024-10-10 15:40:33  
通信连接出现异常,Worker地址:127.0.0.1:54336
[ WARN ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.exceptionCaught(NettyRemotingServer.java:151) ] - 2024-10-10 15:40:33  
通信连接出现异常:java.net.SocketException: Connection reset
[ WARN ] - [ io.netty.channel.DefaultChannelPipeline.onUnhandledInboundException(DefaultChannelPipeline.java:1152) ] - 2024-10-10 15:40:33  
An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
java.net.SocketException: Connection reset
	at java.base/sun.nio.ch.SocketChannelImpl.throwConnectionReset(SocketChannelImpl.java:401)
	at java.base/sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:434)
	at io.netty.buffer.PooledByteBuf.setBytes(PooledByteBuf.java:259)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1132)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:357)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:151)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:1583)
[ INFO ] - [ edu.hitsz.commons.utils.RemotingHelper$1.operationComplete(RemotingHelper.java:74) ] - 2024-10-10 15:40:33  
closeChannel:关闭远程连接 127.0.0.1:54336 true
[ INFO ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.channelInactive(NettyRemotingServer.java:139) ] - 2024-10-10 15:40:33  
客户端断开链接: 127.0.0.1:54336
[ INFO ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.channelUnregistered(NettyRemotingServer.java:123) ] - 2024-10-10 15:40:33  
channel取消注册成功:127.0.0.1:54336
[ WARN ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.exceptionCaught(NettyRemotingServer.java:150) ] - 2024-10-10 15:40:36  
通信连接出现异常,Worker地址:127.0.0.1:54342
[ WARN ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.exceptionCaught(NettyRemotingServer.java:151) ] - 2024-10-10 15:40:36  
通信连接出现异常:java.net.SocketException: Connection reset
[ WARN ] - [ io.netty.channel.DefaultChannelPipeline.onUnhandledInboundException(DefaultChannelPipeline.java:1152) ] - 2024-10-10 15:40:36  
An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
java.net.SocketException: Connection reset
	at java.base/sun.nio.ch.SocketChannelImpl.throwConnectionReset(SocketChannelImpl.java:401)
	at java.base/sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:434)
	at io.netty.buffer.PooledByteBuf.setBytes(PooledByteBuf.java:259)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1132)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:357)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:151)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:1583)
[ INFO ] - [ edu.hitsz.commons.utils.RemotingHelper$1.operationComplete(RemotingHelper.java:74) ] - 2024-10-10 15:40:36  
closeChannel:关闭远程连接 127.0.0.1:54342 true
[ INFO ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.channelInactive(NettyRemotingServer.java:139) ] - 2024-10-10 15:40:36  
客户端断开链接: 127.0.0.1:54342
[ INFO ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.channelUnregistered(NettyRemotingServer.java:123) ] - 2024-10-10 15:40:36  
channel取消注册成功:127.0.0.1:54342
[ WARN ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.exceptionCaught(NettyRemotingServer.java:150) ] - 2024-10-10 15:40:43  
通信连接出现异常,Worker地址:127.0.0.1:54344
[ WARN ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.exceptionCaught(NettyRemotingServer.java:151) ] - 2024-10-10 15:40:43  
通信连接出现异常:java.net.SocketException: Connection reset
[ WARN ] - [ io.netty.channel.DefaultChannelPipeline.onUnhandledInboundException(DefaultChannelPipeline.java:1152) ] - 2024-10-10 15:40:43  
An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
java.net.SocketException: Connection reset
	at java.base/sun.nio.ch.SocketChannelImpl.throwConnectionReset(SocketChannelImpl.java:401)
	at java.base/sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:434)
	at io.netty.buffer.PooledByteBuf.setBytes(PooledByteBuf.java:259)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1132)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:357)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:151)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:1583)
[ INFO ] - [ edu.hitsz.commons.utils.RemotingHelper$1.operationComplete(RemotingHelper.java:74) ] - 2024-10-10 15:40:43  
closeChannel:关闭远程连接 127.0.0.1:54344 true
[ INFO ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.channelInactive(NettyRemotingServer.java:139) ] - 2024-10-10 15:40:43  
客户端断开链接: 127.0.0.1:54344
[ INFO ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.channelUnregistered(NettyRemotingServer.java:123) ] - 2024-10-10 15:40:43  
channel取消注册成功:127.0.0.1:54344
[ INFO ] - [ edu.hitsz.remoting.AbstractRemotingClient.createChannel(AbstractRemotingClient.java:185) ] - 2024-10-10 15:48:29  
开始尝试建立通信连接:localhost:8888
[ INFO ] - [ edu.hitsz.remoting.AbstractRemotingClient.createChannel(AbstractRemotingClient.java:204) ] - 2024-10-10 15:48:29  
建立远程连接成功:localhost:8888,edu.hitsz.remoting.netty.NettyChannelFuture@552f0
[ INFO ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.channelRegistered(NettyRemotingServer.java:116) ] - 2024-10-10 15:48:29  
channel注册成功:127.0.0.1:54469
[ INFO ] - [ edu.hitsz.remoting.AbstractRemotingClient.createChannel(AbstractRemotingClient.java:185) ] - 2024-10-10 15:48:36  
开始尝试建立通信连接:localhost:8888
[ INFO ] - [ edu.hitsz.remoting.AbstractRemotingClient.createChannel(AbstractRemotingClient.java:204) ] - 2024-10-10 15:48:36  
建立远程连接成功:localhost:8888,edu.hitsz.remoting.netty.NettyChannelFuture@32f51b3a
[ INFO ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.channelRegistered(NettyRemotingServer.java:116) ] - 2024-10-10 15:48:36  
channel注册成功:127.0.0.1:54473
[ INFO ] - [ edu.hitsz.remoting.AbstractRemotingClient.createChannel(AbstractRemotingClient.java:185) ] - 2024-10-10 15:48:40  
开始尝试建立通信连接:localhost:8888
[ INFO ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.channelRegistered(NettyRemotingServer.java:116) ] - 2024-10-10 15:48:40  
channel注册成功:127.0.0.1:54474
[ INFO ] - [ edu.hitsz.remoting.AbstractRemotingClient.createChannel(AbstractRemotingClient.java:204) ] - 2024-10-10 15:48:40  
建立远程连接成功:localhost:8888,edu.hitsz.remoting.netty.NettyChannelFuture@67cf8689
[ WARN ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.exceptionCaught(NettyRemotingServer.java:150) ] - 2024-10-10 15:51:15  
通信连接出现异常,Worker地址:127.0.0.1:54474
[ WARN ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.exceptionCaught(NettyRemotingServer.java:151) ] - 2024-10-10 15:51:15  
通信连接出现异常:java.net.SocketException: Connection reset
[ WARN ] - [ io.netty.channel.DefaultChannelPipeline.onUnhandledInboundException(DefaultChannelPipeline.java:1152) ] - 2024-10-10 15:51:15  
An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
java.net.SocketException: Connection reset
	at java.base/sun.nio.ch.SocketChannelImpl.throwConnectionReset(SocketChannelImpl.java:401)
	at java.base/sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:434)
	at io.netty.buffer.PooledByteBuf.setBytes(PooledByteBuf.java:259)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1132)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:357)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:151)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:1583)
[ INFO ] - [ edu.hitsz.commons.utils.RemotingHelper$1.operationComplete(RemotingHelper.java:74) ] - 2024-10-10 15:51:15  
closeChannel:关闭远程连接 127.0.0.1:54474 true
[ INFO ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.channelInactive(NettyRemotingServer.java:139) ] - 2024-10-10 15:51:15  
客户端断开链接: 127.0.0.1:54474
[ INFO ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.channelUnregistered(NettyRemotingServer.java:123) ] - 2024-10-10 15:51:15  
channel取消注册成功:127.0.0.1:54474
[ WARN ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.exceptionCaught(NettyRemotingServer.java:150) ] - 2024-10-10 15:51:20  
通信连接出现异常,Worker地址:127.0.0.1:54469
[ WARN ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.exceptionCaught(NettyRemotingServer.java:151) ] - 2024-10-10 15:51:20  
通信连接出现异常:java.net.SocketException: Connection reset
[ WARN ] - [ io.netty.channel.DefaultChannelPipeline.onUnhandledInboundException(DefaultChannelPipeline.java:1152) ] - 2024-10-10 15:51:20  
An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
java.net.SocketException: Connection reset
	at java.base/sun.nio.ch.SocketChannelImpl.throwConnectionReset(SocketChannelImpl.java:401)
	at java.base/sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:434)
	at io.netty.buffer.PooledByteBuf.setBytes(PooledByteBuf.java:259)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1132)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:357)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:151)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:1583)
[ INFO ] - [ edu.hitsz.commons.utils.RemotingHelper$1.operationComplete(RemotingHelper.java:74) ] - 2024-10-10 15:51:20  
closeChannel:关闭远程连接 127.0.0.1:54469 true
[ INFO ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.channelInactive(NettyRemotingServer.java:139) ] - 2024-10-10 15:51:20  
客户端断开链接: 127.0.0.1:54469
[ INFO ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.channelUnregistered(NettyRemotingServer.java:123) ] - 2024-10-10 15:51:20  
channel取消注册成功:127.0.0.1:54469
[ WARN ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.exceptionCaught(NettyRemotingServer.java:150) ] - 2024-10-10 15:51:28  
通信连接出现异常,Worker地址:127.0.0.1:54473
[ WARN ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.exceptionCaught(NettyRemotingServer.java:151) ] - 2024-10-10 15:51:28  
通信连接出现异常:java.net.SocketException: Connection reset
[ WARN ] - [ io.netty.channel.DefaultChannelPipeline.onUnhandledInboundException(DefaultChannelPipeline.java:1152) ] - 2024-10-10 15:51:28  
An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
java.net.SocketException: Connection reset
	at java.base/sun.nio.ch.SocketChannelImpl.throwConnectionReset(SocketChannelImpl.java:401)
	at java.base/sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:434)
	at io.netty.buffer.PooledByteBuf.setBytes(PooledByteBuf.java:259)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1132)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:357)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:151)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:1583)
[ INFO ] - [ edu.hitsz.commons.utils.RemotingHelper$1.operationComplete(RemotingHelper.java:74) ] - 2024-10-10 15:51:28  
closeChannel:关闭远程连接 127.0.0.1:54473 true
[ INFO ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.channelInactive(NettyRemotingServer.java:139) ] - 2024-10-10 15:51:28  
客户端断开链接: 127.0.0.1:54473
[ INFO ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.channelUnregistered(NettyRemotingServer.java:123) ] - 2024-10-10 15:51:28  
channel取消注册成功:127.0.0.1:54473
[ WARN ] - [ io.netty.bootstrap.AbstractBootstrap.setChannelOption(AbstractBootstrap.java:464) ] - 2024-10-10 16:06:59  
Unknown channel option 'SO_KEEPALIVE' for channel '[id: 0x7c903898]'
[ ERROR ] - [ edu.hitsz.remoting.netty.NettyRemotingServer.serverStart(NettyRemotingServer.java:78) ] - 2024-10-10 16:06:59  
Netty通信服务Server端启动失败!
java.net.BindException: Address already in use: bind
	at java.base/sun.nio.ch.Net.bind0(Native Method)
	at java.base/sun.nio.ch.Net.bind(Net.java:565)
	at java.base/sun.nio.ch.ServerSocketChannelImpl.netBind(ServerSocketChannelImpl.java:344)
	at java.base/sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:301)
	at io.netty.channel.socket.nio.NioServerSocketChannel.doBind(NioServerSocketChannel.java:141)
	at io.netty.channel.AbstractChannel$AbstractUnsafe.bind(AbstractChannel.java:562)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.bind(DefaultChannelPipeline.java:1334)
	at io.netty.channel.AbstractChannelHandlerContext.invokeBind(AbstractChannelHandlerContext.java:600)
	at io.netty.channel.AbstractChannelHandlerContext.bind(AbstractChannelHandlerContext.java:579)
	at io.netty.channel.DefaultChannelPipeline.bind(DefaultChannelPipeline.java:973)
	at io.netty.channel.AbstractChannel.bind(AbstractChannel.java:260)
	at io.netty.bootstrap.AbstractBootstrap$2.run(AbstractBootstrap.java:356)
	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:569)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:1583)
[ INFO ] - [ edu.hitsz.remoting.AbstractRemoting$RemotingEventExecutor.run(AbstractRemoting.java:502) ] - 2024-10-10 16:06:59  
RemotingEventExecutor service started
[ WARN ] - [ io.netty.bootstrap.AbstractBootstrap.setChannelOption(AbstractBootstrap.java:464) ] - 2024-10-10 16:50:23  
Unknown channel option 'SO_KEEPALIVE' for channel '[id: 0xd9870eed]'
[ INFO ] - [ edu.hitsz.remoting.AbstractRemoting$RemotingEventExecutor.run(AbstractRemoting.java:502) ] - 2024-10-10 16:50:23  
RemotingEventExecutor service started
[ WARN ] - [ io.netty.bootstrap.AbstractBootstrap.setChannelOption(AbstractBootstrap.java:464) ] - 2024-10-10 22:43:16  
Unknown channel option 'SO_KEEPALIVE' for channel '[id: 0x924567b1]'
[ INFO ] - [ edu.hitsz.remoting.AbstractRemoting$RemotingEventExecutor.run(AbstractRemoting.java:502) ] - 2024-10-10 22:43:16  
RemotingEventExecutor service started
[ INFO ] - [ edu.hitsz.remoting.AbstractRemotingClient.createChannel(AbstractRemotingClient.java:185) ] - 2024-10-10 22:51:01  
开始尝试建立通信连接:localhost:8888
[ INFO ] - [ edu.hitsz.remoting.AbstractRemotingClient.createChannel(AbstractRemotingClient.java:204) ] - 2024-10-10 22:51:01  
建立远程连接成功:localhost:8888,edu.hitsz.remoting.netty.NettyChannelFuture@cfb7b14
[ INFO ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.channelRegistered(NettyRemotingServer.java:116) ] - 2024-10-10 22:51:01  
channel注册成功:127.0.0.1:61544
[ INFO ] - [ edu.hitsz.remoting.AbstractRemotingClient.createChannel(AbstractRemotingClient.java:185) ] - 2024-10-10 22:51:56  
开始尝试建立通信连接:localhost:8888
[ INFO ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.channelRegistered(NettyRemotingServer.java:116) ] - 2024-10-10 22:51:56  
channel注册成功:127.0.0.1:61561
[ INFO ] - [ edu.hitsz.remoting.AbstractRemotingClient.createChannel(AbstractRemotingClient.java:204) ] - 2024-10-10 22:51:56  
建立远程连接成功:localhost:8888,edu.hitsz.remoting.netty.NettyChannelFuture@2afb7a52
[ INFO ] - [ edu.hitsz.remoting.AbstractRemotingClient.createChannel(AbstractRemotingClient.java:185) ] - 2024-10-10 22:59:49  
开始尝试建立通信连接:localhost:8888
[ INFO ] - [ edu.hitsz.remoting.netty.NettyRemotingServer$NettyConnectManageHandler.channelRegistered(NettyRemotingServer.java:116) ] - 2024-10-10 22:59:49  
channel注册成功:127.0.0.1:61649
[ INFO ] - [ edu.hitsz.remoting.AbstractRemotingClient.createChannel(AbstractRemotingClient.java:204) ] - 2024-10-10 22:59:49  
建立远程连接成功:localhost:8888,edu.hitsz.remoting.netty.NettyChannelFuture@299e0e2f
